dslink-scala-nest/src/main/scala/com/uralian/nest/service/NestHttpService.scala

```diff
@@ -92,4 +103,25 @@ class NestHttpService(client: NestHttpClient) extends NestService {
     client.httpGet[T](s"devices/thermostats/$deviceId/$name").recover {
       case NonFatal(e) => throw new IllegalArgumentException(s"Error retrieving devices/thermostats/$deviceId/$name", e)
     }
+
+  /**
+    * Read a stream of data and send it to sink
+    *
+    * @param token
+    * @param ec
+    * @param system
+    * @tparam T
+    * @return
+    */
+  def readThermostatStream[T]()(implicit token: AccessToken, ec: ExecutionContext,
+                                system: ActorSystem): Future[Source[StreamThermostatData, Any]] = {
+    implicit val materializer = ActorMaterializer()
+    implicit val dataSerializer = ThermostatStreaming.dataSerializer
+    val uri = "/devices/thermostats/"
+    for {
+      source <- client.httpGetStream[StreamThermostatData]("devices/thermostats/")
+    } yield {
+      source
+    }
```

Comment:
Mmm... do we need the whole "for" thing? :) Also, since the method returns thermostat data, I guess we won't need the T type parameter anymore?